### PR TITLE
fix(STRK-52): Numista re-sync allows removing existing on-item tags

### PR DESCRIPTION
**Pull request by lbruton**
> **Draft — QA preview.** Merge to `dev` after review.

## Summary

- Replaces `cb.disabled = true` with `cb.dataset.onItem = "1"` in the Numista tag picker render so on-item tags are shown checked **and enabled** — users can uncheck them to record an explicit removal.
- Uncheck-all gains a `!cb.dataset.onItem` guard, preventing accidental mass-removal of tags already on the item (mirrors the existing `!cb.disabled` guard for blacklisted tags — asymmetric by design; see approach.md D-2).
- Fill Fields now walks all unchecked on-item checkboxes outside the existing checked-tag block, resolves stored-exact-case via `getItemTags()`, and calls `removeItemTag()` (which already calls `addRemovedTag()` internally — not called a second time).

## Test 3 — Spec Correction (NOT a TDD bypass)

**Test 3 in `numista-picker-tags.spec.js` was rewritten as a spec correction** per approach.md D-5. The prior test asserted `toBeDisabled()`, which encoded the *wrong* behavior that STRK-52 explicitly corrects. This is the legitimate "spec was wrong → test correction in the same PR" path called out in CLAUDE.md. Four new AC tests (3-AC3, 3-AC6, 3-AC10, 3-AC5) were added following TDD: written first (red phase), then implementation made them green.

## Test plan

- [x] AC-1 — `3. existing on-item tag is shown checked and enabled with '(on item)' hint` passes
- [x] AC-2 — checked on-item tag leaves item unchanged (no removal recorded)
- [x] AC-3 — `3-AC3: unchecking an on-item tag records removal and opts-out future syncs` passes
- [x] AC-4 — manual-only tags untouched (verified by full suite; picker only iterates Numista tagList)
- [x] AC-5 — `3-AC5: matching manual and Numista tag renders as one checked on-item row` passes
- [x] AC-6 — `3-AC6: unchecking removes the stored tag case-insensitively` passes
- [x] AC-7 — re-checking a removed tag clears opt-out (existing test 10 still passes)
- [x] AC-8 — scalar field preserve-on-unchecked behavior unchanged (existing tests still pass)
- [x] AC-9/10 — `3-AC10: Uncheck-all protects on-item and blacklisted rows simultaneously` passes
- [x] Full offline suite: 524 passed, 0 failed

## Links

- Issue: https://plane.lbruton.cc/lbruton/browse/STRK-52/
- Sketch: `DocVault/Projects/StakTrakr/sketches/STRK-52-numista-resync-tag-membership/`
- Version bump PR: #1094

## Summary by CodeRabbit

* **New Features**
  * Tags already present on inventory items are now clearly marked.
  * Users can opt out of marked tags, with removals properly recorded and persisted across sessions.

* **Tests**
  * Expanded test coverage for tag opt-out workflows and edge cases.

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ Foundation docs: `/Volumes/DATA/GitHub/DocVault/Projects/StakTrakr/Foundation/`
 | Data Pipelines    | `/Volumes/DATA/GitHub/DocVault/Projects/StakTrakr/Foundation/data-pipelines.md`    | Spot / retail / goldback / image pipelines — cron, thresholds, failure modes |
 | Cloud Sync        | `/Volumes/DATA/GitHub/DocVault/Projects/StakTrakr/Foundation/cloud-sync.md`        | Dropbox OAuth, AES-256-GCM, atomic rollback, backup/restore                  |
 
-Tier 2 — 31 deep-dive docs at `DocVault/Projects/StakTrakr/`. Full list: `_Index.md`.
+Tier 2 — 11 deep-dive reference docs at `DocVault/Projects/StakTrakr/Foundation/Deep Dives/`. Linked from Foundation summaries and `Overview.md`.
```
[the remark this answers is not in the document]
Comment (lbruton) [reply CLAUDE.md:33]: **False positive — no change needed.** `CLAUDE.md` is a local Claude Code configuration file, not a source file. The `Overview.md` reference at line 33 (`Linked from Foundation summaries and Overview.md`) refers to a file in the external DocVault vault, not to a file in this repository. This is an external cross-repo reference intentionally documented here for developer orientation. No file needs to be created in StakTrakr.

File: js/catalog-api.js [hunk 2]
```diff
@@ -2402,6 +2405,26 @@ const fillFormFromNumistaResult = () => {
...
+      // Resolve stored-exact-case via case-insensitive search so removeItemTag() matches.
+      const numistaLabel = (cb.dataset.tag || "").toLowerCase();
+      const storedTags = typeof getItemTags === "function" ? getItemTags(_fillUuid) : [];
+      const exactStored = storedTags.find((t) => t.toLowerCase() === numistaLabel);
+      if (exactStored && typeof removeItemTag === "function") {
+        removeItemTag(_fillUuid, exactStored); // internally calls addRemovedTag
+        if (typeof window._renderEditTags === "function") window._renderEditTags();
+      }
+    });
+  }
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/catalog-api.js:2432]: **Resolved in** `de0acedf` — `_renderEditTags()` was hoisted outside the `forEach` loop using a `didRemove` dirty flag; it now fires at most once after all removals complete. `getItemTags()` remains in-loop by design (it re-reads after each `removeItemTag()` mutation, which is functionally correct for this path where N is typically 0–2).

[the remark this answers is not in the document]
Comment (lbruton) [reply js/catalog-api.js:2432]: **Resolved in** `de0acedf` — `_renderEditTags()` was hoisted outside the loop with a `didRemove` dirty flag (single re-render after all removals). `getItemTags()` intentionally remains in-loop: each `removeItemTag()` call mutates the in-memory tag list synchronously, so reading fresh after each removal is correct and ensures subsequent iterations find the remaining tags with accurate casing. The loop is bounded by the number of on-item Numista candidates (typically 0–2).

File: CLAUDE.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lbruton) [reply CLAUDE.md:33]: **Duplicate — already resolved in this review cycle.** See earlier thread on same file/line. False positive — `CLAUDE.md` is a local Claude Code configuration file; the `Overview.md` reference points to an external DocVault vault file, not a file in this repository.

File: js/catalog-api.js [hunk 3]
```diff
@@ -2402,6 +2405,32 @@ const fillFormFromNumistaResult = () => {
...
   }
 
+  // STRK-52: Walk unchecked on-item checkboxes and record explicit opt-outs.
+  // This runs outside the tagCheckboxes block so it fires even when zero tags are checked.
+  // applyNumistaTags() only adds — it never removes existing tags — so getItemTags() still
+  // returns the pre-submit stored list here, making the case-insensitive lookup safe.
+  if (_fillUuid) {
+    const allTagCbs = container.querySelectorAll('input[name="numistaTag"]');
+    let didRemove = false;
+    allTagCbs.forEach((cb) => {
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/catalog-api.js]: **Acknowledged — no change needed.** `_renderEditTags()` was already hoisted outside the loop in `de0acedf` (dirty-flag pattern). `getItemTags()` intentionally remains in-loop: `removeItemTag()` mutates the in-memory tag map synchronously, so re-reading after each removal ensures subsequent iterations see the current list with accurate casing. A pre-loop snapshot would be equivalent but adds indirection without a correctness benefit. Loop is bounded to 0–2 iterations in practice.

File: js/catalog-api.js [hunk 4]
```diff
@@ -2402,6 +2405,32 @@ const fillFormFromNumistaResult = () => {
...
+          : (console.warn("STRK-52: getItemTags not available — skipping on-item removal walk"),
+            []);
+      const exactStored = storedTags.find((t) => t.toLowerCase() === numistaLabel);
+      if (exactStored && typeof removeItemTag === "function") {
+        removeItemTag(_fillUuid, exactStored); // internally calls addRemovedTag
+        didRemove = true;
+      }
+    });
+    if (didRemove && typeof window._renderEditTags === "function") window._renderEditTags();
+  }
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/catalog-api.js]: **Acknowledged — no change needed.** The two `_renderEditTags()` calls serve orthogonal code paths: the first fires when the user checked new tags (inside the `applyNumistaTags` block); the second fires when the user unchecked on-item tags (the STRK-52 removal walk). Merging them would require restructuring the entire submit handler. A double render in the edge case where both paths execute is idempotent — it re-renders from current state — so this is acceptable overhead.

File: js/catalog-api.js [hunk 5]
```diff
@@ -2402,6 +2405,32 @@ const fillFormFromNumistaResult = () => {
...
+    allTagCbs.forEach((cb) => {
+      if (cb.dataset.onItem !== "1" || cb.checked) return;
+      // Tag was on the item at render time but the user unchecked it — record opt-out.
+      // Resolve stored-exact-case via case-insensitive search so removeItemTag() matches.
+      const numistaLabel = (cb.dataset.tag || "").toLowerCase();
+      const storedTags =
+        typeof getItemTags === "function"
+          ? getItemTags(_fillUuid)
+          : (console.warn("STRK-52: getItemTags not available — skipping on-item removal walk"),
+            []);
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/catalog-api.js]: **Fixed in `fbac2a94` — hoisted `getItemTags(_fillUuid)` snapshot above the `forEach`; simplified fallback to an `if/else` guard outside the loop.**
